caffe2/core/hip/miopen_wrapper.h

```diff
@@ -26,8 +26,8 @@ struct MIOPENWorkspace
         if(nbytes_ < nbytes)
         {
             reset();
-            auto data_and_deleter = HIPContext::New(nbytes);
-            data_                 = {data_and_deleter.first, data_and_deleter.second};
+            auto data_ptr = HIPContext::New(nbytes);
+            data_ = {data_ptr.move_context().release(), data_ptr.get_deleter()};
```

Comment:
If you move out the context, doesn't that also move out the deleter? In that case you're going to get the wrong deleter in the code here.

The root cause of the problem is that you should store a DataPtr. I think that's worth doing in this PR.

Reply:
I was worried to break things in the beginning, but looking at APIs for this one, we should be able to do that.